includes/functions.php

```diff
@@ -1116,3 +1116,47 @@ function enrich_content_data( $content, $regex, $regex_callback ) {
 
 	return $content_with_links;
 }
+
+function generate_post_summary( $post, $length = 500 ) {
+	$post = get_post( $post );
+
+	if ( ! $post ) {
+		return '';
+	}
+
+	$content = \sanitize_post_field( 'post_excerpt', $post->post_excerpt, $post->ID );
+
+	if ( $content ) {
+		return $content;
+	}
+
+	$content       = \sanitize_post_field( 'post_content', $post->post_content, $post->ID );
+	$content_parts = \get_extended( $content );
+
+	$excerpt_more = \apply_filters( 'activitypub_excerpt_more', '[&hellip;]' );
+	$length       = $length - strlen( $excerpt_more );
```

Comment:
Wouldn’t this be 500 - 10, since it counts characters, whereas `[&hellip;]` in the end only represents 3 characters ([…])?

Reply:
🤔

so should we use `[...]` instead?

Reply:
Should be working fine. Entities should only be used if the correct sign cannot. So, as long as you use the correct hellip sign (it’s a single one: `…`), this should be fine and the correct count for `strlen`